prov/efa/src/rdm/rxr_ep.h

```diff
@@ -279,6 +285,49 @@ static inline int rxr_need_sas_ordering(struct rxr_ep *ep)
 	return ep->msg_order & FI_ORDER_SAS;
 }
 
+static inline uint64_t rxr_get_host_id(char *host_id_file)
+{
+	FILE *fp = NULL;
+	const size_t HOST_ID_LENGTH = 16;
+	char host_id_str[HOST_ID_LENGTH + 1];
+	char *end_ptr = NULL;
+	const size_t HOST_ID_PREFIX_LENGTH = 3; /* Ignore the i-0 sequence */
+	size_t length = 0;
+	uint64_t host_id = 0;
+
+	if (!host_id_file) {
+		EFA_WARN(FI_LOG_EP_CTRL, "Host id file is not specified\n");
+		goto out;
+	}
+
+	fp = fopen(host_id_file, "r");
+	if (!fp) {
+		EFA_WARN(FI_LOG_EP_CTRL, "Cannot open host id file: %s\n", host_id_file);
+		goto out;
+	}
+
+	fseek(fp, HOST_ID_PREFIX_LENGTH, SEEK_SET);
+
+	length = fread(host_id_str, 1, HOST_ID_LENGTH, fp);
+	host_id_str[HOST_ID_LENGTH + 1] = '\0';
```

Comment:
should be `host_id_str[HOST_ID_LENGTH] = '\0'`

Reply:
Also, I prefer move this line to after L316

Reply:
Thank you! Fixed.